### PR TITLE
CIRC-331:  Library and  Location

### PR DESCRIPTION
**Pull request by VRohach**
## Purpose

CIRC-331: Add library location fields:
 - New repository added for '/libraries' endpoint.
 - New entity for 'Library' added.
 - New call to LibrariesRepository was added to the 'RequestRepository'.
 - New fields 'code' and 'libraryName' were added to the 'RequestRepresentation' for adding them in response for this ednpoint: '/circulation/requests'.
 - Tests for these new fields were changed.

### CODE REVIEW

**Comment by mreno-EBSCO** [comment]
@VRohach could you update the PR description with something about the work done in this PR? I can read the JIRA, but I think some documentation here is important, even if it is just an overview of the commit comments. Thanks!

**Comment by VRohach** [comment]
> @VRohach could you update the PR description with something about the work done in this PR? I can read the JIRA, but I think some documentation here is important, even if it is just an overview of the commit comments. Thanks!

Yes, sure. I added this one. Thank you!

File: src/main/java/org/folio/circulation/domain/RequestRepository.java [hunk 1]
```diff
@@ -26,6 +26,7 @@
   private final LoanRepository loanRepository;
   private final ServicePointRepository servicePointRepository;
   private final PatronGroupRepository patronGroupRepository;
+  private final LibrariesRepository librariesRepository;
   
 
   private RequestRepository(
```
[the remark this answers is not in the document]
Comment (VRohach) [reply src/main/java/org/folio/circulation/domain/RequestRepository.java]: @mreno-EBSCO Ok, I will add this SuppressWarning here.

File: src/main/java/org/folio/circulation/domain/LibrariesRepository.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    MultipleRecords<Request> multipleRequests) {
+    Collection<Request> requests = multipleRequests.getRecords();
+
+    final List<String> librariesToFetch =
+      multipleRequests.getRecords()
+        .stream()
+        .filter(Objects::nonNull)
+        .map(Request::getItem)
+        .map(Item::getLibraryId)
+        .collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (VRohach) [reply src/main/java/org/folio/circulation/domain/LibrariesRepository.java]: @mreno-EBSCO  Good point. I will fix it!

File: src/main/java/org/folio/circulation/domain/Library.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    return getProperty(representation, "name");
+  }
+
+  public String getCode() {
+    return getProperty(representation, "code");
+  }
+
+  public String getCampusId() {
+    return getProperty(representation, "campusId");
+  }
```
[the remark this answers is not in the document]
Comment (VRohach) [reply src/main/java/org/folio/circulation/domain/Library.java]: @mreno-EBSCO Yes, sure. I will remove unused methods

File: src/main/java/org/folio/circulation/domain/Item.java [hunk 4]
```diff
@@ -230,6 +230,10 @@ else if(holding != null && holding.containsKey(PERMANENT_LOCATION_ID)) {
     }
   }
 
+  public String getLibraryId(){
+    return  getProperty(getLocation(),"libraryId");
```
[the remark this answers is not in the document]
Comment (VRohach) [reply src/main/java/org/folio/circulation/domain/Item.java]: @mreno-EBSCO Of course, thanks for this comment.

File: src/main/java/org/folio/circulation/domain/LibrariesRepository.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    final MultipleRecordFetcher<Library> fetcher = createLibrariesFetcher();
+
+    return fetcher.findByIds(librariesToFetch)
+      .thenApply(r -> r.next(multiRequests -> {
+          Map<String, Library> result = multiRequests.getRecords()
+            .stream()
+            .collect(Collectors.toMap(Library::getId, Function.identity()));
+          for (Request request : requests) {
+           request.getItem().getLocation().put("libraryName",
```
[the remark this answers is not in the document]
Comment (VRohach) [reply src/main/java/org/folio/circulation/domain/LibrariesRepository.java]: @mreno-EBSCO Ok, I will try to make this code more clear.

**Pull request closed without merging**